src/elements/content-sidebar/activity-feed/task-form/TaskForm.js

```diff
@@ -102,6 +102,41 @@ function convertAssigneesToSelectorItems(approvers: Array<TaskCollabAssignee>):
     });
 }
 
+function getEditErrorMessage(taskType: TaskType, editMode?: TaskEditMode, error?: { status: number }): React.Node {
+    const isEditMode = editMode === TASK_EDIT_MODE_EDIT;
+    const taskErrorMessage = isEditMode ? messages.taskUpdateErrorMessage : apiMessages.taskCreateErrorMessage;
+    const isForbiddenErrorOnEdit = error && error.status && error.status === 403 && isEditMode;
+
+    if (!error) {
+        return null;
+    }
+
+    switch (taskType) {
+        case 'GENERAL':
+            return isForbiddenErrorOnEdit ? (
+                <InlineError title={<FormattedMessage {...messages.taskEditWarningTitle} />}>
+                    <FormattedMessage {...messages.taskGeneralAssigneeRemovalWarningMessage} />
+                </InlineError>
+            ) : (
+                <InlineError title={<FormattedMessage {...messages.taskCreateErrorTitle} />}>
+                    <FormattedMessage {...taskErrorMessage} />
+                </InlineError>
+            );
+        case 'APPROVAL':
+            return isForbiddenErrorOnEdit ? (
```

Comment:
Inline error component is repeated. You can just set the title message and tile body conditionally in the switch/blocks, and use 1 component at the ned